File: erykandroidcommon/src/main/java/pl/edu/amu/wmi/erykandroidcommon/rx/RxUtils.java
package pl.edu.amu.wmi.erykandroidcommon.rx;

import android.content.Context;
import android.preference.PreferenceManager;

import com.f2prateek.rx.preferences2.RxSharedPreferences;

import lombok.NonNull;

/**
 * @author Eryk Mariankowski <eryk.mariankowski@softra.pl> on 19.07.17.
 */
public class RxUtils {

    private RxUtils() {
    }

    public static RxSharedPreferences getSharedPreferences(@NonNull Context mContext) {
        return RxSharedPreferences.create(
                PreferenceManager.getDefaultSharedPreferences(mContext));
    }

    public static RxSharedPreferences getSharedPreferences(@NonNull Context mContext, @NonNull String prefsName) {
        return RxSharedPreferences.create(
                mContext.getSharedPreferences(prefsName, Context.MODE_PRIVATE));
    }


}

File: erykandroidcommon/src/main/java/pl/edu/amu/wmi/erykandroidcommon/rx/RxUtils.kt
package pl.edu.amu.wmi.erykandroidcommon.rx

import android.content.Context
import android.preference.PreferenceManager

import com.f2prateek.rx.preferences2.RxSharedPreferences

import lombok.NonNull

/**
 * @author Eryk Mariankowski <eryk.mariankowski></eryk.mariankowski>@softra.pl> on 19.07.17.
 */
internal object RxUtils {

    fun getSharedPreferences(@NonNull mContext: Context): RxSharedPreferences {
        return RxSharedPreferences.create(
                PreferenceManager.getDefaultSharedPreferences(mContext))
    }

    fun getSharedPreferences(@NonNull mContext: Context, @NonNull prefsName: String): RxSharedPreferences {
        return RxSharedPreferences.create(
                mContext.getSharedPreferences(prefsName, Context.MODE_PRIVATE))
    }


}
